x-pack/plugins/ml/public/jobs/new_job/simple/components/utils/validate_job.js

```diff
@@ -50,6 +50,12 @@ export function populateValidationMessages(validationResults, checks) {
     checks.groupIds.message = msg;
   }
 
+  if (validationResults.contains('model_memory_limit_units_invalid')) {
+    checks.modelMemoryLimit.valid = false;
+    const msg = `Model memory limit data unit unrecognized. It must be B, KB, MB, GB, TB or PB`;
```

Comment:
As above, the units referenced in the message should come from `ALLOWED_DATA_UNITS `.